.circleci/cimodel/pytorch_build_definitions.py

```diff
@@ -0,0 +1,369 @@
+import cimodel.dimensions as dimensions
+import cimodel.conf_tree as conf_tree
+from cimodel.conf_tree import ConfigNode
+import cimodel.visualization as visualization
+
+
+DOCKER_IMAGE_PATH_BASE = "308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/"
+
+DOCKER_IMAGE_VERSION = 282
+
+
+class DockerHide(object):
+    """
+    Used for hiding name elements for construction of the Docker image path.
+    Name elements that are wrapped in this object may be part of the build configuration name, but
+    shall be excluded from the Docker path.
+    """
+    def __init__(self, val):
+        self.val = val
+
+    def __str__(self):
+        return self.val
+
+
+class Conf(object):
+    def __init__(self,
+                 distro,
+                 parms,
+                 pyver=None,
+                 cuda_version=None,
+                 is_xla=False,
+                 restrict_phases=None,
+                 gpu_resource=None,
+                 dependent_tests=None,
+                 parent_build=None):
+
+        self.distro = distro
+        self.pyver = pyver
+        self.parms = parms
+        self.cuda_version = cuda_version
+        self.is_xla = is_xla
+        self.restrict_phases = restrict_phases
+        self.gpu_resource = gpu_resource
+        self.dependent_tests = dependent_tests or []
+        self.parent_build = parent_build
+
+    def get_parms(self):
+        leading = ["pytorch"]
+        if self.is_xla:
+            leading.append(DockerHide("xla"))
+
+        cuda_parms = []
+        if self.cuda_version:
+            cuda_parms.extend(["cuda" + self.cuda_version, "cudnn7"])
+        return leading + ["linux", self.distro] + cuda_parms + self.parms
+
+    # TODO: Eliminate this special casing in docker paths
+    def gen_docker_image_path(self):
+
+        build_env_pieces = list(map(str, filter(lambda x: type(x) is not DockerHide, self.get_parms())))
+        base_build_env_name = "-".join(build_env_pieces)
+
+        return miniutils.quote(DOCKER_IMAGE_PATH_BASE + base_build_env_name + ":" + str(DOCKER_IMAGE_VERSION))
+
+    def get_build_job_name_pieces(self, build_or_test):
+        return self.get_parms() + [build_or_test]
+
+    def gen_build_name(self, build_or_test):
+        return ("_".join(map(str, self.get_build_job_name_pieces(build_or_test)))).replace(".", "_")
+
+    def get_dependents(self):
+        return self.dependent_tests
+
+    def gen_yaml_tree(self, build_or_test):
+
+        build_job_name_pieces = self.get_build_job_name_pieces(build_or_test)
+
+        build_env_name = "-".join(map(str, build_job_name_pieces))
+
+        env_dict = OrderedDict([
+            ("BUILD_ENVIRONMENT", build_env_name),
+            ("DOCKER_IMAGE", self.gen_docker_image_path()),
+        ])
+
+        if self.pyver:
+            env_dict["PYTHON_VERSION"] = miniutils.quote(self.pyver)
+
+        if build_or_test == "test" and self.cuda_version:
+            env_dict["USE_CUDA_DOCKER_RUNTIME"] = miniutils.quote("1")
```

Comment:
This is the right logic for using the cuda runtime and should be used for the caffe2 jobs too. This is actually also true for all the binary jobs